pages/wiki/openembedded.md

```diff
@@ -3,10 +3,243 @@ title: OpenEmbedded
+meta-smelt            /home/developer/asteroid/src/meta-smartwatch/meta-smelt  7
+meta-sparrow          /home/developer/asteroid/src/meta-smartwatch/meta-sparrow  7
+meta-sprat            /home/developer/asteroid/src/meta-smartwatch/meta-sprat  7
+meta-sturgeon         /home/developer/asteroid/src/meta-smartwatch/meta-sturgeon  7
+meta-swift            /home/developer/asteroid/src/meta-smartwatch/meta-swift  8
+meta-tetra            /home/developer/asteroid/src/meta-smartwatch/meta-tetra  7
+meta-wren             /home/developer/asteroid/src/meta-smartwatch/meta-wren  7
+```
+
+By convention, Yocto layers all begin with `meta`. There are layers for individual pieces such as Qt, networking, Python, and filesystems. These are followed by the layers for each of the watch codenames. If you are looking for an AsteroidOS-specific application, you can probably guess that it's in the `meta-asteroid` layer, but you can also ask the computer to search for you.
+
+Originally, `bitbake` and its related tools were the only way to do things, but there is now a newer and much more convenient tool named [`devtool`](https://docs.yoctoproject.org/dev/ref-manual/devtool-reference.html). To search for a recipe with `flashlight` in the name:
+
+```
+devtool search flashlight
+```
+
+After a minute or so, `devtool` finds a single recipe: `asteroid-flashlight`. You can now modify this recipe:
+
+
+```
+devtool modify asteroid-flashlight
+```
+
+The last two lines of output will be something like these:
+
+```
+INFO: Source tree extracted to /home/developer/asteroid/build/workspace/sources/asteroid-flashlight
+INFO: Recipe asteroid-flashlight now set up to build from /home/developer/asteroid/build/workspace/sources/asteroid-flashlight
+
+```
+Navigate into this directory, and find the relevant line in `src/main.qml` and alter the color from white to light blue:
+
+```
+color: flashOn ? "lightblue" : "#66444444"
+```
+
+Save the altered source code and build this copy of the flashlight app:
+
+```
+devtool build asteroid-flashlight
+```
+
+After some time, the build will complete and you can test the application on a watch.
+
+```
+devtool deploy-target asteroid-flashlight root@192.168.2.15
+```
+
+Now the flashlight app will show a light blue screen. Reverting to the original version is very simple:
+
+```
+devtool undeploy-target asteroid-flashlight root@192.168.2.15
+```
+
+# Adding a New Application
+
+---
+
+Let's say you have decided to create a new application. As described in [Creating an Asteroid app]({{rel 'wiki/creating-an-asteroid-app'}}) page, the [Asteroid-helloworld](https://github.com/AsteroidOS/asteroid-helloworld) which can be used as the basis for your own new application can be cloned and then modified. So let's say you've done that and now you'd like to build the app, deploy it and test it. You can, of course, follow the instructions listed on that page, but here we'll show how to do so by working within Yocto and bitbake.
```

Comment:
"So let's say you've done that and now you'd like to build the app, deploy it and test it. You can, of course, follow the instructions listed on that page, but here we'll show how to do so by working within Yocto and bitbake."

I would reword that to:
"So let's say you've done that and now you'd like to integrate the app, to the Yocto and bitbake build system"

The intent here shouldn't be to provide two different ways to do the same thing just for the sake of it but rather a way to get changes upstream :)

Reply:
better!   done.